bgpd/bgp_nht.c

```diff
@@ -911,3 +911,52 @@ void bgp_nht_register_enhe_capability_interfaces(struct peer *peer)
 						BGP_UNNUM_DEFAULT_RA_INTERVAL);
 	}
 }
+
+void bgp_nht_dereg_enhe_capability_intfs(struct peer *peer)
+{
+	struct bgp *bgp;
+	struct bgp_node *rn;
+	struct bgp_nexthop_cache *bnc;
+	struct nexthop *nhop;
+	struct interface *ifp;
+	struct prefix p;
+
+	if (peer->ifp)
+		return;
+
+	bgp = peer->bgp;
+
+	if (!bgp->nexthop_cache_table[AFI_IP6])
+		return;
+
+	if (!sockunion2hostprefix(&peer->su, &p)) {
+		if (BGP_DEBUG(nht, NHT))
+			zlog_debug("%s: Unable to convert sockunion to prefix",
+				   __PRETTY_FUNCTION__);
+		return;
+	}
+
+	if (p.family != AF_INET6)
+		return;
+
+	rn = bgp_node_lookup(bgp->nexthop_cache_table[AFI_IP6], &p);
+	if (!rn)
+		return;
+
+	bnc = bgp_node_get_bgp_nexthop_info(rn);
+	if (!bnc)
```

Comment:
wouldn't it be useful to have a debug log here as well? I meant for every test in this function.

Reply:
I can add more debugs, but since they don't exist for the register function, I didn't think them necessary for the deregister function.  I'm fine adding them if you prefer.

Reply:
For me, that's ok how it is. I'm just looking into the source and thinking how to see that behavior if having a problem in the future.